impersonate_login/tests/test_impersonate_login.py

```diff
@@ -13,8 +13,36 @@ class TestImpersonateLogin(HttpCase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
+        # Admin always exists; ensure it can impersonate and has a known password
         cls.admin_user = cls.env.ref("base.user_admin")
-        cls.demo_user = cls.env.ref("base.user_demo")
+        # Make sure admin can use the feature (group check in session info)
+        group_imp = cls.env.ref("impersonate_login.group_impersonate_login")
+        if group_imp not in cls.admin_user.group_ids:
+            cls.admin_user.group_ids += group_imp
+        # Ensure a demo user exists for tests on Odoo 19 (no demo loaded by default)
+        try:
+            cls.demo_user = cls.env.ref("base.user_demo")
+        except Exception:  # ValueError when not found
```

Comment:
May as well catch ValueError then. Or maybe even better, don't rely on the demo user at all in this test as the code to create a custom user is right below (but with a different login)